readthedocs/api/v2/client.py

```diff
@@ -41,7 +40,7 @@ def setup_api():
         connect=3,
         status=3,
         backoff_factor=0.5,  # 0.5, 1, 2 seconds
-        method_whitelist=('GET', 'PUT', 'PATCH', 'POST'),
+        method_whitelist=('GET', 'PUT', 'PATCH'),
```

Comment:
This defaults to https://urllib3.readthedocs.io/en/latest/reference/urllib3.util.html#urllib3.util.retry.Retry.DEFAULT_METHOD_WHITELIST seem like nice defaults, except I'd add patch too.

Reply:
I do think we want to retry POST on some status codes, if there is support for it. For example, a 502 we likely don't want to retry, but I think we do on a 503. I think this is better than nothing if we can't customize that, but it would be great to have more control. 

It seems `status_forcelist` is likely what we want, and to continue to include POST in this.

Reply:
`RETRY_AFTER_STATUS_CODES` maybe is actually what we want.

Reply:
So, I omitted the 504 response (Gateway Timeout). Not sure about `502 Bad Gateway` https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#5xx_Server_errors

But anyway, when it times out with post, it doesn't return any status code (it times out bc it hasn't responded yet!). So using that won't prevent from triggering the task multiple times.

Reply:
Good point re: timeout.